vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/PropertyKindMap.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2011-2022 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.sqlclient.impl;
+
+import io.vertx.sqlclient.PropertyKind;
+
+import java.util.Arrays;
+
+/**
+ * A map-like storage for {@link PropertyKind} instances.
+ * It is suitable for our clients which store very few instances (usually just one).
+ */
+public final class PropertyKindMap {
+
+  private Object[] elements;
+  private int count;
+
+  public PropertyKindMap() {
+    elements = new Object[2];
+    count = 0;
+  }
+
+  public Object get(PropertyKind<?> property) {
```

Comment:
```suggestion
  public Object get(PropertyKind<?> property) {
      final Object[] elements = this.elements;
      for (int i = 0, size = elements.length; i < size; i+=2) {
         final Object key = elements[i];
         assert key != null : "key cannot be null"; 
         if (key == property || key.equals(property)) {
             return elements[i + 1];
         }  
      }
      return null;
  }
```
This will help JIT to correlate the loop variables with the length of the array, saving bound checks and inlining the equality check would help in case key is effectively the same (if it's not common, you can drop the ref equality check)

Reply:
I just pushed 8620798d34eb9f84942f2a13f78bfa33d3d4831d to remove the computed `count` field. At the same time, I've changed the loop to use the `elements.length` in condition check. Can you verify it's enough or is still better to make a local copy of the reference to the array?